### PR TITLE
Ohai support

### PR DESCRIPTION
**Pull request by erikng**
This matches Wes' work for ohai. We still have to null values since ohai is dumb.

### CODE REVIEW

File: postflight.d/sal-postflight [hunk 1]
```diff
@@ -283,41 +283,40 @@ def get_puppet_report():
...
 
-def get_ohai_report():
+def get_ohai_report(insert_name):
     report = None
     ohai_path = '/opt/chef/bin/ohai'
+    ohai = dict()
+    new_ohai = dict()
     if os.path.exists(ohai_path):
-        command = [ohai_path, '-L', '/usr/local/sal/ohaiwarnings.log']
+        command = ['/opt/chef/bin/ohai']
```
Comment (grahamgilbert) [inline postflight.d/sal-postflight]: Should we use `ohai_path` here?

File: postflight.d/sal-postflight [hunk 2]
```diff
@@ -283,41 +283,40 @@ def get_puppet_report():
...
+    return hashrocket_flatten_dict(new_ohai)
 
 
 def nullify(ohai):
-    # Temp fix while we we wait for Chef to approve Ohai PR.
+    # Ohai still returns None values. Doubtful it will be fixed.
     for key in ohai:
         if ohai[key] is None:
-            ohai[key] = "null"
+            ohai[key] = ''
```
Comment (grahamgilbert) [inline postflight.d/sal-postflight]: Maybe just delete them? I'm not sure of the value of sending empty strings.

File: postflight.d/sal-postflight [hunk 3]
```diff
@@ -283,41 +283,40 @@ def get_puppet_report():
...
         try:
             report = subprocess.check_output(command)
         except subprocess.CalledProcessError as error:
             munkicommon.display_debug2('Issue getting ohai report:')
             munkicommon.display_debug2(error.message)
-
-    ohai = {}
     if report:
         try:
             ohai = json.loads(report)
```
Comment (weswhet) [inline postflight.d/sal-postflight]: changing this to `json.loads(report, object_pairs_hook=utils.dict_clean)` would probably solve your skip facts issue.

**Comment by erikng** [comment]
with @weswhet idea, this is now functioning without the null function I wrote.

**Comment by clburlison** [comment]
For what it's worth. This works for me.

**Comment by weswhet** [comment]
LGTM

**Pull request merged**